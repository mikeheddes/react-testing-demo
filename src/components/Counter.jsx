import React, { Component } from "react";
import styled from "styled-components";

import { Wrapper, Label, Button, Line } from "./Other";

const StyledButton = styled(Button)`
  width: 50%;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 0;

  &:first-of-type {
    text-align: left;
    left: 0;
    right: 50%;
  }

  &:last-of-type {
    text-align: right;
    left: 50%;
    right: 0;
  }
`;

const StyledLine = styled(Line)`
  position: absolute;
  left: calc(50% - 5px / 2);
  height: 28px;

  &:first-of-type {
    top: 0;
  }

  &:last-of-type {
    bottom: 0;
  }
`;

class Counter extends Component {
  state = { value: 0 };

  handleDecreaseCount = () => {
    this.setState({ value: this.state.value - 1 });
  };

  handleIncreaseCount = () => {
    this.setState({ value: this.state.value + 1 });
  };

  render() {
    const { value } = this.state;
    return (
      <Wrapper color="#eb4b4b">
        <StyledButton data-testid="dec" onClick={this.handleDecreaseCount}>
          -
        </StyledButton>
        <Label>{value}</Label>
        <StyledButton data-testid="inc" onClick={this.handleIncreaseCount}>
          +
        </StyledButton>
        <StyledLine />
        <StyledLine />
      </Wrapper>
    );
  }
}

export default Counter;
