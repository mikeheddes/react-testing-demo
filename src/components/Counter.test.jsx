import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Counter from "./Counter";

describe("<Counter />", () => {
  // it("should set the state to state + 1 when button is pressed", () => {
  //   const wrapper = shallow(<Counter />);
  //   expect(wrapper.state("value")).toBe(0);
  //   wrapper.find("button").simulate("click");
  //   expect(wrapper.state("value")).toBe(1);
  // });

  it("should set the state to state + 1 when button is pressed", () => {
    const testInstance = renderer.create(<Counter />).root;
    expect(testInstance.instance.state.value).toBe(0);
    testInstance.instance.handleIncreaseCount();
    expect(testInstance.instance.state.value).toBe(1);
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
