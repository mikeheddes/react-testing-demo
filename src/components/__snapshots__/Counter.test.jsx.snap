// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should match snapshot 1`] = `
.c0 {
  background-color: #eb4b4b;
  position: relative;
  box-shadow: 8px 8px rgba(0,0,0,0.15);
}

.c2 {
  display: block;
  text-align: center;
  color: white;
  font-weight: 500;
  font-size: 42px;
  padding: 39px 50px;
}

.c1 {
  color: white;
  font-weight: 500;
  font-size: 42px;
  padding: 39px 50px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  outline: none;
  background-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  width: 50%;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 0;
}

.c1:active {
  opacity: 0.5;
}

.c1:first-of-type {
  text-align: left;
  left: 0;
  right: 50%;
}

.c1:last-of-type {
  text-align: right;
  left: 50%;
  right: 0;
}

.c3 {
  width: 5px;
  background-color: white;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: absolute;
  left: calc(50% - 5px / 2);
  height: 28px;
}

.c3:first-of-type {
  top: 0;
}

.c3:last-of-type {
  bottom: 0;
}

<div
  className="c0"
  color="#eb4b4b"
>
  <button
    className="c1"
    id="dec"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="c2"
  >
    0
  </span>
  <button
    className="c1"
    id="inc"
    onClick={[Function]}
  >
    +
  </button>
  <div
    className="c3"
  />
  <div
    className="c3"
  />
</div>
`;
